tests/browser.js

```diff
@@ -401,9 +407,88 @@ module.exports = {
       })
     })
     this.controller.getServices().router.trigger()
+  },
+
+  'should run redirect async': function (test) {
+    test.expect(0)
+    this.controller.signals({
+      'existing': [
+        function checkAction () {
+          test.ok(true)
+        }
+      ],
+      'missing': [
+        redirect('/existing', {replace: false})
+      ]
+    })
+
+    this.controller.modules({
+      devtools: function () {},
+      router: Router({
+        '/existing': 'existing',
+        '/*': 'missing'
+      })
+    })
+    this.controller.getServices().router.trigger()
+    test.done()
+  },
```

Comment:
A bit unsure if we catch the scenario here. I think it is safer to run "test.done()" on an action after the redirect, inside missing. Just to be perfectly sure :)